js/modules/k6/http/response_callback_test.go

```diff
@@ -0,0 +1,444 @@
+						"expected_response": "false", // this is on purpose
+						"proto":             "HTTP/1.1",
+					},
+					metrics: allHTTPMetrics,
+				},
+				{
+					tags: map[string]string{
+						"method":            "GET",
+						"url":               sr("HTTPBIN_URL/get"),
+						"name":              sr("HTTPBIN_URL/get"),
+						"status":            "200",
+						"group":             "",
+						"expected_response": "true",
+						"proto":             "HTTP/1.1",
+					},
+					metrics: allHTTPMetrics,
+				},
+			},
+		},
+
+		"global overwrite": {
+			code: `http.request("GET", "HTTPBIN_URL/redirect/1", null, {responseCallback: http.expectedStatuses(200)});`,
+			expectedSamples: []expectedSample{
+				{
+					tags: map[string]string{
+						"method":            "GET",
+						"url":               sr("HTTPBIN_URL/redirect/1"),
+						"name":              sr("HTTPBIN_URL/redirect/1"),
+						"status":            "302",
+						"group":             "",
+						"expected_response": "false", // this is on purpose
+						"proto":             "HTTP/1.1",
+					},
+					metrics: allHTTPMetrics,
+				},
+				{
+					tags: map[string]string{
+						"method":            "GET",
+						"url":               sr("HTTPBIN_URL/get"),
+						"name":              sr("HTTPBIN_URL/get"),
+						"status":            "200",
+						"group":             "",
+						"expected_response": "true",
+						"proto":             "HTTP/1.1",
+					},
+					metrics: allHTTPMetrics,
+				},
+			},
+		},
+		"per request overwrite with null": {
+			code: `http.request("GET", "HTTPBIN_URL/redirect/1", null, {responseCallback: null});`,
+			expectedSamples: []expectedSample{
+				{
+					tags: map[string]string{
+						"method": "GET",
+						"url":    sr("HTTPBIN_URL/redirect/1"),
+						"name":   sr("HTTPBIN_URL/redirect/1"),
+						"status": "302",
+						"group":  "",
+						"proto":  "HTTP/1.1",
+					},
+					metrics: HTTPMetricsWithoutFailed,
+				},
+				{
+					tags: map[string]string{
+						"method": "GET",
+						"url":    sr("HTTPBIN_URL/get"),
+						"name":   sr("HTTPBIN_URL/get"),
+						"status": "200",
+						"group":  "",
+						"proto":  "HTTP/1.1",
+					},
+					metrics: HTTPMetricsWithoutFailed,
+				},
+			},
+		},
+		"global overwrite with null": {
+			code: `
+			http.setResponseCallback(null);
+			res = http.request("GET", "HTTPBIN_URL/redirect/1");
+			`,
+			expectedSamples: []expectedSample{
+				{
+					tags: map[string]string{
+						"method": "GET",
+						"url":    sr("HTTPBIN_URL/redirect/1"),
+						"name":   sr("HTTPBIN_URL/redirect/1"),
+						"status": "302",
+						"group":  "",
+						"proto":  "HTTP/1.1",
+					},
+					metrics: HTTPMetricsWithoutFailed,
+				},
+				{
+					tags: map[string]string{
+						"method": "GET",
+						"url":    sr("HTTPBIN_URL/get"),
+						"name":   sr("HTTPBIN_URL/get"),
+						"status": "200",
+						"group":  "",
+						"proto":  "HTTP/1.1",
+					},
+					metrics: HTTPMetricsWithoutFailed,
+				},
+			},
+		},
```

Comment:
Please add a `http.batch()` test as well. There shouldn't be any concurrency issues right now, given how restricted the callbacks are (still :heart: how you managed to find that workaround!), but I'd like us to have such a test now, before we add any more functionality here.

Reply:
db01bba